### PR TITLE
Send uid to elastic

### PR DESCRIPTION
**Pull request by davidpmccormick**
For [#140](https://github.com/wellcomecollection/content-api/issues/140)

## What does this change?

It gets the `uid` from the Prismic document for `Article`s and `Events` and sends it along in the Elasticsearch transformers.

## How to test

- `cd pipeline`
- `yarn test`

## How can we measure success?

Gets us closer to being able to use `uid`s instead of `id`s in our urls (relates to [11074](https://github.com/wellcomecollection/wellcomecollection.org/issues/11074))

## Have we considered potential risks?

I'm new here!

### CODE REVIEW

File: pipeline/src/transformers/eventDocument.ts [hunk 1]
```diff
@@ -166,6 +167,7 @@ export const transformEventDocument = (
 
   return {
     id,
+    uid,
     ...(tags.includes('delist') && { isChildScheduledEvent: true }),
     display: {
```
Comment (rcantin-w) [inline pipeline/src/transformers/eventDocument.ts:172]: Should the UID be added here as well?

File: pipeline/src/types/transformed/index.ts [hunk 2]
```diff
@@ -46,6 +46,7 @@ export type {
 
 export type ElasticsearchArticle = {
   id: string;
+  uid: string | null;
```
[the remark this answers is not in the document]
Comment (rcantin-w) [reply pipeline/src/types/transformed/index.ts:49]: Also, it shouldn't really happen but it has the potential too (e.g. we add the uid field to existing documents)

**Comment by rcantin-w** [comment]
Left a few comments but wanted to say it otherwise looked good and nice to have the snapshots updated 👍 

This can be tested once in staging too, using our toggles and accessing the staging API, so we have that extra safety check before deploying to prod.

**Comment by kenoir** [comment]
Do the index mappings also need updating to ensure we can index the new UID field properly?

File: pipeline/src/transformers/article.ts [hunk 3]
```diff
@@ -87,6 +87,7 @@ export const transformArticle = (
   document: ArticlePrismicDocument
 ): ElasticsearchArticle => {
   const { data, id, first_publication_date: firstPublicationDate } = document;
+  const uid = document.uid || undefined;
```
Comment (agnesgaroux) [inline pipeline/src/transformers/article.ts:90]: Can it not be destructured like the other `document` attributes?

File: pipeline/src/transformers/eventDocument.ts [hunk 4]
```diff
@@ -143,6 +143,7 @@ export const transformEventDocument = (
     id,
     tags,
   } = document;
+  const uid = document.uid || undefined;
```
Comment (agnesgaroux) [inline pipeline/src/transformers/eventDocument.ts:146]: Ditto

**Comment by agnesgaroux** [comment]
LGTM! 👍 What's the rationale for having the uuid at the document's root level? Would it work to only have it in the `display` object?

**Review by agnesgaroux — APPROVED** [review]

**Pull request merged**